### ISSUE TITLE
False positives in syntax validation (Valid Class Field Syntax is rejected by the linter)

### ISSUE THREAD

**Issue by Skeletonxf** [mozilla/addons-linter#2758]
### Describe the problem and steps to reproduce it:

I was trying to update my addon Duplicate Tab on AMO and ran into two false positives in the linting stage. I have been manually testing the addon in Firefox Developer Edition and am completely sure everything is working without syntax issues.

This is the zip I am trying to upload on AMO. At the time of writing my [repository](https://github.com/Skeletonxf/duplicate-tab) has identical source code.
[duplicate-tab.zip](https://github.com/mozilla/addons-linter/files/3525447/duplicate-tab.zip)

### What happened?

Two false positives were given.

> JavaScript syntax error
> Error: There is a JavaScript syntax error in your code; validation cannot continue on this file.
background/duplication.js line 1 column 1

https://github.com/Skeletonxf/duplicate-tab/blob/089bb703aed2b2013d875caf00bb05cdf6fd7b6a/background/duplication.js#L1-L3
```js
import core from '/core/script.js'
import defaults from '/settings/defaults.js'

```

I assume, though haven't tested, that this error might be due to me not using semicolons and a discrepancy between when the linter expects semicolons and when actual JavaScript interpreters do.

> JavaScript syntax error
> Error: There is a JavaScript syntax error in your code; validation cannot continue on this file.
core/script.js line 65 column 9

https://github.com/Skeletonxf/duplicate-tab/blob/089bb703aed2b2013d875caf00bb05cdf6fd7b6a/core/script.js#L62-L69

```js
    /*
     * Simple logging utility that is turned off on release
     */
    log = (msg) => { //line 65
        if (this.isDebug) {
            console.log(msg)
        }
    }
```

This false positive is more problematic for me, as I had written this method using arrow functions so that `this` is correct when I call it in other places. This is not the only arrow function method in my codebase either.

### What did you expect to happen?

The included zip file to pass validation

### Anything else we should know?

I ran the validation check approximately at  2019/08/21 13:00 UTC on the online linter on AMO.

**Comment by Skeletonxf** [comment]
The second false positive seems to be related to Class Fields https://discourse.mozilla.org/t/javascript-syntax-error-in-addon-validation-results/46195/2?u=skeletonxf

**Comment by Mubelotix** [comment]
Same for me. I am using wasm pack. I can't sign my extension...

**Comment by Mubelotix** [comment]
That issue should stay open

**Comment by jetwitaussi** [comment]
I also have this problem, unable to publish firefox addon because arrow function not suppored. everything is working in firefox, this bug is very annoying. 
Was able to publish on chrome store, so mozilla and their validation are out of date!

**Comment by Skeletonxf** [comment]
Both errors were in fact be due to [class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields) which have been supported by Firefox since version 69. [Replacing the class fields](https://github.com/Skeletonxf/duplicate-tab/commit/066449f834e3510ce3dde7af7059e4395887f334) that were both valid JavaScript and correctly interpreted when debugging the addon in my `core/script.js` file removed the second error but the first remained.

Today I tried commenting out the imports in my `background/duplication.js` file (where the other linter error was reporting a syntax problem) which revealed a new 'error' reported by the linter in the same file that was also using class fields I had missed the first time.

Refactoring those as well to use older syntax finally passed the addons linter, and reintroducing the imports did not add any new errors.

The linter very unhelpfully was reporting a syntax error in my imports due to the completely unrelated reason that I was using class fields further down in the same file.

As annoying as this is, refactoring class fields out of your addon isn't that hard.

```js
class Foo {
  bar = () => console.log('linter needs an update')
}
```

can be transformed into

```js
class Foo {
  constructor() {
    this.bar = () => console.log('linter needs an update')
  }
}
```

I've finally been able to publish my new version after removing the last use of class fields I had missed the first time but to the unhelpful linter errors.

Could @jetwitaussi and @Mubelotix try making this same syntax refactor and see if the linter will accept the updated versions of your code?

**Comment by Mubelotix** [comment]
I can't try your solution because I totally changed the structure of my project. I was making an extension running wasm code and the issue was in the js file containing the glue code, which was auto-generated.

**Comment by willdurand** [comment]
duplicate of https://github.com/mozilla/addons-linter/issues/3062

**Comment by Skeletonxf** [comment]
I opened this issue many months before that one?
